backend/core/socketio_config.py

```diff
@@ -138,16 +144,82 @@ def save_chat_history(
         ).first()
 
         if not conversation_exist:
+            # TODO :: handle this correctly
             return None
 
+        # user/officer message mark as READ
+        new_chat_status = Chat_Status_Enum.READ
+
+        # get message template lang
+        message_template_lang = generate_message_template_lang_by_phone_number(
+            phone_number=client_phone_number
+        )
+        send_conversation_reconnect_template = check_24h_window(
+            session=session, chat_session_id=conversation_exist.id
+        )
+        # Send conversation reconnect template if beyond 24hr
+        if (
+            platform == Platform_Enum.WHATSAPP.value
+            and send_conversation_reconnect_template
+        ):
+            # user/officer message mark as READ
+            # TODO ::
+            # new_chat_status = Chat_Status_Enum.AWAITING_USER_REPLY
+
+            # get message template ID
+            content_sid = os.getenv(
+                f"CONVERSATION_RECONNECT_TEMPLATE_{message_template_lang}"
+            )
+            # send conversation reconnect template
+            client = session.exec(
+                select(Client).where(
+                    Client.phone_number == client_phone_number
+                )
+            ).first()
+            # farmer name
+            client_name = (
+                client.properties.name
+                if client and client.properties
+                else client_phone_number
+            )
+            # save conversation reconnect template chat to database
+            # get message template from twilio
+            template_content = get_template_content_from_json(
+                content_sid=content_sid
+            )
+            conversation_reconnect_message = f"Hi {client_name},\n"
+            conversation_reconnect_message += "Please reply this message"
+            conversation_reconnect_message += " to restart your conversation."
+            if template_content and not TESTING:
+                conversation_reconnect_message = template_content.replace(
+                    "{{1}}", client_name
```

Comment:
For know, there's no encapsulation message, we will do that in different ticket as we discussed in refinement meeting.

Saving the chat history is only a step to save the chat into our database and this will not delivered into Farmer yet. So we need the message template to send to reach the Farmer.

Reply:
Hmm, I think I need to read the code more but I've run out of time this week before my leave. This looks to me like `whatsapp_message_template_create` here will cause the template to be sent to the farmer and appear in whatsapp on their phone. Is this correct or does it do something else?

If a function called `save_chat_history` has a side effect of sending a templated message, that seems strange to me. What does saving a record of the chat to our DB have to do with sending the reconnect template? 

I'm expecting to see a function call to send a message and in that function there is some logic to work out whether to send the message or encapsulate it in a template and send that. Then when that succeeds the chat history is saved based on that the function actually sent.

I understand that this PR is not trying to do encapsulation yet but in that case I would expect that this function decides whether to send the message or a static template (or maybe it decides to do nothing if we want to send the template based on some different user input and just do nothing if we know the message won't be delivered) and then saves whatever it sent into the history after it succeeds in sending the message.

Like I say, I think I need to read the code properly to understand!

Reply:
Sorry Jacob, 

I understand, will do the homework in other PR that I already started to do (split the function, etc I will do that in other PR). But the step here is we save the message to DB first the sent it into the farmer. We can create another endpoint to track the [message status from Twilio](https://www.twilio.com/docs/messaging/guides/track-outbound-message-status), but that will not send back the message content, so why we need to save the message first. 

For now we doesn't implement the status tracking yet. I believe we will need to do a refactor to our table structure if we need to track the message status using Twilio API.